### ISSUE TITLE
Unable to resolve repository dependencies from network path in Gradle 4.4

### ISSUE THREAD

**Issue by BPaasch** [gradle/gradle#4002]
Gradle 4.4 is unable to resolve dependencies that are located on a network share.
This works with Gradle 4.3

### Expected Behavior
Gradle should find the dependencies in the network share according to the repository definition.

### Current Behavior

Gradle fails with an exception
Could not resolve all files for configuration ':compileClasspath'.

### Context

I'm trying to build a project with dependencies that are on a network share in my organization.

### Steps to Reproduce (for bugs)

[gradle-test.zip](https://github.com/gradle/gradle/files/1615443/gradle-test.zip)

1. From the attached archive, extract the contents.
2. Map the ivy folder to a network share
3. Update the build.gradle script to reference the network share (ex: //my-computer-name/ivy)
4. Run ./gradlew build

### Your Environment

* Build scan URL: https://scans.gradle.com/s/cllnzw5ykwuwg/
* Windows 7
* Git Bash

**Comment by oehme** [comment]
@jjohannes is this something you could look into?

**Comment by jjohannes** [comment]
Confirmed. Thanks for reporting @BPaasch. Running against a repository folder that does not exist reveals the issue in the error message.

**4.3.1**
```
  Searched in the following locations:
      file:////VBOXSVR/missing/ivy/javax.jms/jms/1.1/ivy-1.1.xml
```
**4.4.1**
```
  Searched in the following locations:
      file:///VBOXSVR/missing/ivy/javax.jms/jms/1.1/ivy-1.1.xml
```
One `/` is missing in 4.4.1. Also if you prefix the url with more `////` the results stay the same. There must be a change in url parsing that caused this. And we obviously miss a test case.
I will dig up the corresponding change and fix.

**Comment by BPaasch** [comment]
@jjohannes thanks for looking into this.  Will this be fixed in a 4.4.x release (perhaps 4.4.2)?

**Comment by oehme** [comment]
Not 4.4.x (4.5 is due tomorrow), but we might hold up 4.5 for this, depending on whether there is any workaround.

**Comment by jjohannes** [comment]
I found the cause: e13e45a08759242b56006f8a2b6ce46753434b2c

I have a fix ready - https://github.com/gradle/gradle/compare/jjohannes/fix-4002
But I want to add more test coverage.

This is only reproducible on Windows. I don't think there is a workaround.

**Comment by jjohannes** [comment]
@BPaasch the fix will be in 4.5. We are going to release 4.5 RC2 shortly. Please give it a try to confirm that the fix works for you.

**Comment by BPaasch** [comment]
I was able to test this using 4.5 RC2 and it works.  Thanks for fixing this.
